front-end/src/app/components/timer/timer.component.ts

```diff
@@ -9,20 +8,25 @@ import { AppComponent } from "../../app.component";
   styleUrls: ["./timer.component.css", "../../../assets/css/main.css"]
 })
 export class TimerComponent implements OnInit {
+  constructor(private app: AppComponent) {}
+
   private subscription: Subscription;
   displayTime: string;
   everySecond: Observable<number> = timer(0, 1000);
 
-  constructor(private app: AppComponent) {}
-
   ngOnInit() {
     this.subscription = this.everySecond.subscribe(seconds => {
-      if (this.app.timeState === "stateActive") {
-        this.app.remainingTime = this.app.remainingTime - 1000;
-        this.displayTime = formatMS(this.app.remainingTime);
-      } else if (this.app.timeState === "stateIdle") {
-        this.displayTime = formatMS(this.app.remainingTime);
+      for (const t of this.app.timerList.getAll().values()) {
```

Comment:
maybe rename `t` to `timer` so the code is easier to understand

Reply:
timer is already taken though